pkg/flowaggregator/flowaggregator.go

```diff
@@ -724,3 +818,130 @@ func (fa *flowAggregator) createInfoElementForTemplateSet(ieName string, enterpr
 	}
 	return ie, nil
 }
+
+// updateIfConfigChanges has a watcher to detect if modification happens. Then check whether we should update clickhouse or flow-collector at runtime
+func (fa *flowAggregator) updateIfConfigChanges(stopCh <-chan struct{}) {
+	if fa.configData == nil {
+		var err error
+		fa.configData, err = ioutil.ReadFile(fa.configFile)
+		if err != nil {
+			klog.ErrorS(err, "Get flow aggregator configData failed:")
+		}
+	}
+	go wait.NonSlidingUntil(fa.watchConfigmap, 5*time.Second, stopCh)
+}
+
+func (fa *flowAggregator) watchConfigmap() {
+	klog.Info("Watching for FA configuration file")
+	defer klog.Info("Stopping watch on FA configuration file")
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		klog.ErrorS(err, "Error when creating watcher")
+	}
+	err = watcher.Add(fa.configFile)
+	if err != nil {
+		klog.ErrorS(err, "Error when adding watched file")
+	}
+	for {
+		select {
+		case event, ok := <-watcher.Events:
+			if !ok {
+				return
+			}
+			if event.Op == fsnotify.Remove {
+				watcher.Add(fa.configFile)
+				break
+			}
```

Comment:
I assume this means that the file has been deleted? This seems like a weird way to handle it.
If the file has been removed (which is unlikely, someone would need to mess with it manually), then calling `watcher.Add(fa.configFile)` would fail because the file doesn't exist. Then what's the expected behavior?

Reply:
This is because the fsnotify will stop watching if the mounted configMap updated due to our changes.
There are two events: CHMOD and REMOVE. Both of them will happened when mounted configMap updated.
Watcher will stop if REMOVE event happens (so I need to add the same file back to watcher)